src/client/opamPinCommand.ml

```diff
@@ -571,32 +571,28 @@ and source_pin
     | _ -> opam_opt
   in
 
-  if not need_edit && opam_opt = None then
-    OpamConsole.note
-      "No package definition found for %s: please complete the template"
-      (OpamConsole.colorise `bold (OpamPackage.to_string nv));
-
-  let need_edit = need_edit || opam_opt = None in
-
   let opam_opt =
-    let opam_base = match opam_opt with
-      | None -> OpamFileTools.template nv
-      | Some opam -> opam
-    in
-    let opam_base =
-      OpamFile.OPAM.with_url_opt urlf opam_base
-    in
     if need_edit then
-      (if not (OpamFile.exists temp_file) then
-         OpamFile.OPAM.write_with_preserved_format
-           ?format_from:(OpamPinned.orig_opam_file st name opam_base)
-           temp_file opam_base;
-       edit_raw name temp_file >>|
-       (* Preserve metadata_dir so that copy_files below works *)
-       OpamFile.OPAM.(with_metadata_dir (metadata_dir opam_base))
-      )
+      let opam_base = match opam_opt with
+        | None ->
+          OpamConsole.note
+            "No package definition found for %s: please complete the template"
+            (OpamConsole.colorise `bold (OpamPackage.to_string nv));
+          OpamFileTools.template nv
```

Comment:
do we really want to keep this template around? In which situation is it useful? I feel like it doesn't really has any purpose and we might want to remove it entirely (probably in a separate PR)